21.md

```diff
@@ -0,0 +1,209 @@
+LUD-21: Currencies in `payRequest`.
+=================================
+
+`author: lsunsi`
+`author: luizParreira`
+`author: lorenzolfm`
+
+---
+
+## Support for LNURL-pay currencies
+
+This document describes an extension to the [payRequest](https://github.com/lnurl/luds/blob/luds/06.md) base specification that allows the `WALLET` to send money to a `SERVICE` denominating the amount in a different currencies. The main use cases supported by this extension are sending sats denominating the value in a currency with an exchange rate defined by the `SERVICE` and optionally converting the value into the currency upon payment of the invoice. Widespread implementation of this extension by broker-enabled wallets would enable a remittance-like experience for users around the globe.
+
+The extension is opt-in and backwards compatible. Further, a supporting `WALLET` can always tell if a `SERVICE` is also supporting beforehand so the communication is never ambiguous.
+
+### Wallet-side first request
+
+The first request is unchanged from the base specification.
+
+### Service-side first response
+
+`SERVICE` must alter its JSON response to the first callback to include a `currencies` field, as follows:
+
+```typescript
+type BaseResponse = {
+    tag: "payRequest",
+    metadata: string,
+    callback: string,
+    maxSendable: number,
+    minSendable: number
+};
+
+type Currency = {
+  code: string, // Code of the currency, used as an ID for it. E.g.: BRL
+  name: string, // Name of the currency. E.g.: Reais
+  symbol: string, // Symbol of the currency. E.g.: R$
+  decimals: number, // Number of decimal places. E.g.: 2
+  multiplier: number, // Number of millisatoshis per smallets unit of currency. E.g.: 5405.405
```

Comment:
When converting the base currency to milisats or sats, should it be rounded away from zero or rounded towards zero?

Reply:
Great question. I think the conversion at this point can be anything (away, towards or anything) because the multiplier works as a volatile hint of the actual conversion rate. The actual guaranteed (for limited time) conversion rate is given by the callback call only. Does that make sense to you?

Reply:
@lsunsi yup, makes lots of sense. thanks for clarifying